src/site/stages/build/plugins/check-broken-links/helpers/getBrokenLinks.js

```diff
@@ -0,0 +1,45 @@
+const cheerio = require('cheerio');
+
+const _isBrokenLink = require('./isBrokenLink');
+
+function getBrokenLinks(file, allPaths, isBrokenLink = _isBrokenLink) {
+  const $ = cheerio.load(file.contents);
+  const elements = $('a, img');
+  const currentPath = file.path;
+
+  const linkErrors = [];
+
+  elements.each((index, node) => {
+    const $node = $(node);
+
+    let target = null;
+
+    if ($node.is('a')) {
+      const namedAnchor = !!$node.prop('name');
+
+      target = $node.attr('href');
+
+      if (!target && namedAnchor) {
+        return;
+      }
+    }
+
+    if ($node.is('img')) {
+      target = $node.attr('src') || $node.attr('data-src');
+    }
+
+    const isBroken = isBrokenLink(target, currentPath, allPaths);
+
+    if (isBroken) {
+      const html = cheerio.html($node);
+      linkErrors.push({
+        html,
+        target,
+      });
+    }
+  });
+
+  return linkErrors;
+}
+
+module.exports = getBrokenLinks;
```

Comment:
Every HTML file is process through this - it parses the link elements and evaluates the validity of each link value using a helper function `isBrokenLink`. It returns an array of link errors.